pyrit/prompt_target/prompt_shield_target.py

```diff
@@ -0,0 +1,158 @@
+        documents: ['document1', 'document2']
+
+        None is the default state (use parsing). userPrompt and document are the other states, and
+        you can use those to force only one parameter (either userPrompt or documents) to be populated
+        with the raw input (no parsing).
+        '''
+
+    async def send_prompt_async(self, *, prompt_request: PromptRequestResponse) -> PromptRequestResponse:
+        '''
+        For an up-to-date version of the Prompt Shield spec, have a look at the Prompt Shield quickstart guide:
+        https://learn.microsoft.com/en-us/azure/ai-services/content-safety/quickstart-jailbreak
+        Or read the tutorial + documentation notebook.
+        '''
+        self._validate_request(prompt_request=prompt_request)
+
+        request = prompt_request.request_pieces[0]
+
+        logger.info(f"Sending the following prompt to the prompt target: {request}")
+
+        headers = {
+            'Ocp-Apim-Subscription-Key': self._api_key,
+            'Content-Type': 'application/json',
+        }
+
+        params = {
+            'api-version': '2024-02-15-preview',
+        }
+
+        userPrompt, docsList = self._input_parser(request.original_value)
+
+        body = {
+            'userPrompt': userPrompt,
+            'documents': docsList
+        }
+
+        response = await net_utility.make_request_and_raise_if_error_async(
+            endpoint_uri=f'{self._endpoint}/contentsafety/text:shieldPrompt',
+            method='POST',
+            params=params,
+            headers=headers,
+            request_body=body
+        )
+
+        logger.info("Received a valid response from the prompt target")
+
+        data = response.content
+
+        response_entry = construct_response_from_request(
+            request=request, response_text_pieces=[data.decode('utf-8')], response_type="text"
+        )
+
+        return response_entry
+
+    
+    def _validate_request(self, *, prompt_request: PromptRequestResponse) -> None:
+        if len(prompt_request.request_pieces) == 0:
+            raise ValueError("This target requires at least one prompt request piece.")
+        if len(prompt_request.request_pieces) > 1:
+            raise ValueError(
+                    "Sorry, but requests with multiple entries are not supported. " \
+                    "Please wrap each PromptRequestPiece in a PromptRequestResponse." \
+                )
+        if type(prompt_request.request_pieces[0].original_value) != str:
+            raise ValueError("The content of this request must be a string.")
+        
+    def _input_parser(self, input_str: str) -> tuple[str, list[str]]:
+        '''
+        Parses the input given to the target to extract the two fields sent to
+        Prompt Shield:
+        userPrompt: str
+        and
+        documents: list[str]
+        '''
+
+        match self._force_entry_kind:
+            case 'userPrompt':
+                return (input_str, [""])
```

Comment:
Pythonic: don't need to wrap return statements in braces
```suggestion
                return input_str, [""]
```